numpy/core/numeric.py

```diff
@@ -2441,14 +2444,24 @@ def array_equal(a1, a2, equal_nan=False):
     if a1.shape != a2.shape:
         return False
     if not equal_nan:
-        return bool(asarray(a1 == a2).all())
+        return bool((a1 == a2).all())
+    cannot_have_nan = (_dtype_cannot_hold_nan(a1.dtype) and
+                       _dtype_cannot_hold_nan(a2.dtype))
+    if cannot_have_nan:
+        if a1 is a2:
```

Comment:
How common is the case of comparing identical (as in `id`) arrays? This slows down the other paths (although not a lot, the `a1 is a2` is quite fast)

Reply:
As for many things... it depends on the usage. In napari for example quite a lot (but for small-ish array).
As the model it's event based when something change it regularly check the dimensions orders, masked data and bunch stuff like that to know wether it should redraw the screen.


The `a is b` case is actually the one the that drove me here in the first case.